pkg/collector/checkhelpers/backoff/backoff.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package backoff
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"sync"
+)
+
+const (
+	countdown = "countdown"
+	maximum   = "maximum"
+)
+
+type RetryableOperation[T any] func() (T, error)
```

Comment:
I think you can probably the common `Retrier` with the right settings or another already in-use backoff.